src/Simulation/Simulators/SparseSimulator/Native/basic_quantum_state.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <list>
+
+#include "types.h"
+#include "gates.h"
+
+namespace Microsoft::Quantum::SPARSESIMULATOR
+{
+
+// Virtual class for QuantumState
+// This is not templated, so it allows SparseSimulator types to avoid templates
+class BasicQuantumState 
+{
+public:
+
+    BasicQuantumState() {}
```

Comment:
If you have time and desire you can apply the following changes also, see attached.
[01.diff.txt](https://github.com/microsoft/qsharp-runtime/files/7238795/01.diff.txt)

Reply:
Updated in feature/sparse-simulator branch.